### ISSUE TITLE
Cron.php not working

### ISSUE THREAD

**Issue by santhoshnsscoe** [magento/magento2#21817]
### Preconditions (*)
1. Magento 2.2.7
2. php 7.1.16

### Steps to reproduce (*)
According to https://devdocs.magento.com/guides/v2.2/config-guide/cli/config-cli-subcommands-cron.html we configured the cron jobs. It will create three commands in the crontab.
```
#~ MAGENTO START
#* * * * */5 /usr/local/bin/php /var/www/html/bin/magento cron:run 2>&1 | grep -v "Ran jobs by schedule" >> /var/www/html/var/log/magento.cron.log
#* * * * */5 /usr/local/bin/php /var/www/html/update/cron.php >> /var/www/html/var/log/update.cron.log
#* * * * */5 /usr/local/bin/php /var/www/html/bin/magento setup:cron:run >> /var/www/html/var/log/setup.cron.log
#~ MAGENTO END
```
### Expected result (*)
The crons works and no errors come.

### Actual result (*)
The ```/usr/local/bin/php /var/www/html/update/cron.php``` is giving error.

```
Warning: require_once(/var/www/html/update/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/update/app/bootstrap.php on line 15

Fatal error: require_once(): Failed opening required '/var/www/html/update/vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/update/app/bootstrap.php on line 15
```

Checked the update folder. There is not vendor folder. So, ran composer update. But that is also giving error message, as the php is not compatible.

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - This package requires php ~5.5.0|~5.6.0|~7.0.0 but your PHP version (7.1.16) does not satisfy that requirement.
  Problem 2
    - Installation request for magento/composer 1.0.3 -> satisfiable by magento/composer[1.0.3].
    - magento/composer 1.0.3 requires php ~5.5.0|~5.6.0|~7.0.0 -> your PHP version (7.1.16) does not satisfy that requirement.
```

**Comment by magento-engcom-team** [comment]
Hi @santhoshnsscoe. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@santhoshnsscoe do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @santhoshnsscoe thank you for you report, i'm not able to reproduce following steps you described on clean 2.2.7 and 2.3.0 seems like your update folder have wrong version. did you try update magento or some else ? i'm closing as this is not reproducible.

**Comment by NiklasBr** [comment]
@engcom-backlog-nazar we are also experiencing this, can it be re-opened?

We are running Magento 2.3.1 (Enterprise Edition) in a Docker container and get this in our logs:

```log
[11-Apr-2019 07:08:42 UTC] PHP Warning:  require_once(/var/www/magento/update/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:08:42 UTC] PHP Stack trace:
[11-Apr-2019 07:08:42 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:08:42 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
[11-Apr-2019 07:08:42 UTC] PHP Fatal error:  require_once(): Failed opening required '/var/www/magento/update/vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:08:42 UTC] PHP Stack trace:
[11-Apr-2019 07:08:42 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:08:42 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
```

**Comment by ghost** [comment]
Hi @NiklasBr  please use command cron:install to have correct cron tab settings 
![DeepinScreenshot_select-area_20190411101548](https://user-images.githubusercontent.com/40756439/55937951-e6208a80-5c42-11e9-8179-25f698b51513.png)

**Comment by NiklasBr** [comment]
Gets me this:

```log
Crontab has been generated and saved
==> var/log/system.log <==
[2019-04-11 07:18:42] report.ERROR: crontab -l 2>&1
no crontab for www-data [] []
[2019-04-11 07:18:43] report.INFO: echo "
#~ MAGENTO START 7cd0a74c421051ec6cce8936eb9edd02
* * * * * /usr/local/bin/php /var/www/magento/bin/magento cron:run 2>&1 | grep -v \"Ran jobs by schedule\" >> /var/www/magento/var/log/magento.cron.log
* * * * * /usr/local/bin/php /var/www/magento/update/cron.php >> /var/www/magento/var/log/update.cron.log
* * * * * /usr/local/bin/php /var/www/magento/bin/magento setup:cron:run >> /var/www/magento/var/log/setup.cron.log
#~ MAGENTO END 7cd0a74c421051ec6cce8936eb9edd02
" 2>&1 | crontab - 2>&1
 [] []

==> var/log/php-error.log <==
[11-Apr-2019 07:17:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:17:01 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
[11-Apr-2019 07:18:01 UTC] PHP Warning:  require_once(/var/www/magento/update/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:18:01 UTC] PHP Stack trace:
[11-Apr-2019 07:18:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:18:01 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
[11-Apr-2019 07:18:01 UTC] PHP Fatal error:  require_once(): Failed opening required '/var/www/magento/update/vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:18:01 UTC] PHP Stack trace:
[11-Apr-2019 07:18:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:18:01 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
[11-Apr-2019 07:19:01 UTC] PHP Warning:  require_once(/var/www/magento/update/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:19:01 UTC] PHP Stack trace:
[11-Apr-2019 07:19:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:19:01 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
[11-Apr-2019 07:19:01 UTC] PHP Fatal error:  require_once(): Failed opening required '/var/www/magento/update/vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:19:01 UTC] PHP Stack trace:
[11-Apr-2019 07:19:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
```

**Comment by ghost** [comment]
@NiklasBr as i'm mentioned above i'm not able to reproduce followings steps, this may be related to your configurations, however maybe someone else can reproduce this on clean install, and please if you reopen this issue please describe detail steps to reproduce. thanks.

**Comment by ghost** [comment]
@NiklasBr did you have vendor folder on update ?

**Comment by NiklasBr** [comment]
Not sure what kind of configurations would that be. We just got started with Magento 2.3.1 with an entire new environment, there is no vendor folder in the update one, it is missing it seems.

**Comment by ghost** [comment]
@NiklasBr so in my case i'm have vendor folder, please try to run composer update.

**Comment by NiklasBr** [comment]
Didn't help unfortunately:

```
$ composer update -v
Loading composer repositories with package information
Updating dependencies (including require-dev)                                     
Dependency resolution completed in 8.362 seconds
Analyzed 19865 packages to resolve dependencies
Analyzed 1310424 rules to resolve dependencies
Dependency resolution completed in 0.009 seconds
Package operations: 0 installs, 3 updates, 0 removals
Updates: composer/composer:1.8.5, myclabs/deep-copy:1.9.1, theseer/tokenizer:1.1.2
  - Updating composer/composer (1.8.4 => 1.8.5): Downloading (100%)         
  - Updating myclabs/deep-copy (1.8.1 => 1.9.1): Downloading (100%)         
  - Updating theseer/tokenizer (1.1.0 => 1.1.2): Downloading (100%)         
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Writing lock file
Generating autoload files
```
```log
==> var/log/system.log <==
[2019-04-11 07:26:12] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:27:10] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:28:11] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:29:12] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:30:10] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:31:12] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:32:13] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:33:11] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:34:13] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []
[2019-04-11 07:35:13] report.INFO: Consumer "async.operations.all" skipped as required connection "amqp" is not configured. Unknown connection name amqp [] []

==> var/log/php-error.log <==
[11-Apr-2019 07:35:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:35:01 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
[11-Apr-2019 07:36:01 UTC] PHP Warning:  require_once(/var/www/magento/update/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:36:01 UTC] PHP Stack trace:
[11-Apr-2019 07:36:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:36:01 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
[11-Apr-2019 07:36:01 UTC] PHP Fatal error:  require_once(): Failed opening required '/var/www/magento/update/vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/magento/update/app/bootstrap.php on line 15
[11-Apr-2019 07:36:01 UTC] PHP Stack trace:
[11-Apr-2019 07:36:01 UTC] PHP   1. {main}() /var/www/magento/update/cron.php:0
[11-Apr-2019 07:36:01 UTC] PHP   2. require_once() /var/www/magento/update/cron.php:7
```

**Comment by ghost** [comment]
@NiklasBr have you this file in update folder  ?

> /vendor/autoload.php'

**Comment by NiklasBr** [comment]
```shell
$ ll /var/www/magento
total 1392
-rw-r--r--  1 www-data www-data 434981 Apr 10 12:47 CHANGELOG.md
-rw-r--r--  1 www-data www-data    650 Apr 10 12:47 COPYING.txt
-rw-r--r--  1 www-data www-data   2994 Apr 10 12:47 Gruntfile.js.sample
-rw-r--r--  1 www-data www-data  10364 Apr 10 12:47 LICENSE.txt
-rw-r--r--  1 www-data www-data  10376 Apr 10 12:47 LICENSE_AFL.txt
-rw-r--r--  1 www-data www-data  30634 Apr 10 12:47 LICENSE_EE.txt
-rw-r--r--  1 www-data www-data   2657 Apr 10 12:47 README_EE.md
drwxrwxr-x  5 www-data www-data   4096 Apr 10 12:47 app
-rw-rw-r--  1 www-data www-data    152 Apr 10 12:20 auth.json
-rw-r--r--  1 www-data www-data    234 Apr 10 12:47 auth.json.sample
drwxrwxr-x  2 www-data www-data   4096 Apr 10 12:47 bin
-rw-rw-r--  1 www-data www-data   3827 Apr 10 12:20 composer.json
-rw-rw-r--  1 www-data www-data 837969 Apr 11 07:44 composer.lock
drwxrwxr-x  5 www-data www-data   4096 Apr 10 12:47 dev
drwxrwxr-x  3 www-data www-data   4096 Apr 11 07:45 generated
-rw-r--r--  1 www-data www-data     57 Apr 10 12:47 grunt-config.json.sample
-rw-r--r--  1 www-data www-data   1370 Apr 10 12:47 index.php
drwxrwxr-x  4 www-data www-data   4096 Apr 10 12:47 lib
-rw-r--r--  1 www-data www-data   5625 Apr 10 12:47 nginx.conf.sample
-rw-r--r--  1 www-data www-data   1416 Apr 10 12:47 package.json.sample
drwxr-xr-x  2 www-data www-data   4096 Apr 10 12:47 phpserver
drwxrwxr-x  6 www-data www-data   4096 Apr 10 12:47 pub
drwxrwxr-x  7 www-data www-data   4096 Apr 10 12:47 setup
drwxrwxr-x  6 www-data www-data   4096 Apr  2 09:29 update
drwxrwxr-x 10 www-data www-data   4096 Apr 11 08:04 var
drwxrwxr-x 75 www-data www-data   4096 Apr 11 07:44 vendor
```

```shell
$ ll /var/www/magento/vendor | grep auto     
-rw-r--r--   1 www-data www-data   178 Apr 11 07:44 autoload.php
```

**Comment by ghost** [comment]
@NiklasBr can you show output ? 
 `php -r "define('UPDATER_BP', realpath(__DIR__ . '/../')); echo UPDATER_BP;"`

**Comment by NiklasBr** [comment]
```bash
/var/www
```

**Comment by ghost** [comment]
@NiklasBr ok, you have a point here.
![DeepinScreenshot_select-area_20190411111741](https://user-images.githubusercontent.com/40756439/55941720-795dbe00-5c4b-11e9-9097-74e89ac6efcf.png)

**Comment by ghost** [comment]
@NiklasBr seems like you have no path here -

> > usr/local/lib/php

**Comment by NiklasBr** [comment]
Please elaborate, that directory exists...

**Comment by ghost** [comment]
@NiklasBr file php ? did  you have /usr/local/lib/php ? ` ls -l /usr/local/lib/php`

**Comment by NiklasBr** [comment]
```shell
$ ls -l /usr/local/lib/php
total 124
drwxr-sr-x  2 root staff  4096 Mar 27 01:40 Archive
drwxr-sr-x  2 root staff  4096 Mar 27 01:40 Console
drwxr-sr-x  2 root staff  4096 Mar 27 01:40 OS
drwxr-sr-x 11 root staff  4096 Mar 27 01:40 PEAR
-rw-r--r--  1 root staff 35466 Mar 27 01:40 PEAR.php
drwxr-sr-x  3 root staff  4096 Mar 27 01:40 Structures
-rw-r--r--  1 root staff 20562 Mar 27 01:40 System.php
drwxr-sr-x  2 root staff  4096 Mar 27 01:40 XML
drwxr-sr-x  2 root staff  4096 Mar 27 01:40 build
drwxr-sr-x  3 root staff  4096 Mar 27 01:40 data
drwxr-sr-x  1 root staff  4096 Apr  3 16:21 doc
drwxr-sr-x  1 root staff  4096 Mar 27 01:40 extensions
-rw-r--r--  1 root staff 15176 Mar 27 01:40 pearcmd.php
-rw-r--r--  1 root staff  1025 Mar 27 01:40 peclcmd.php
drwxr-sr-x  5 root staff  4096 Mar 27 01:40 test
```

**Comment by ghost** [comment]
@NiklasBr did you have vendor/autoload.php in core derectory ?

**Comment by NiklasBr** [comment]
@engcom-backlog-nazar [I believe so](#issuecomment-482012931) (if I understand your question correctly).

**Comment by ghost** [comment]
@NiklasBr oh,, and what about autoload in update folder ?

**Comment by NiklasBr** [comment]
No:

```shell
$ ll /var/www/magento/update/
total 116
-rw-rw-r-- 1 www-data www-data     0 Apr  2 09:29 README.md
drwxrwxr-x 3 www-data www-data  4096 Apr 10 06:38 app
-rw-rw-r-- 1 www-data www-data   703 Apr  2 09:29 composer.json
-rw-rw-r-- 1 www-data www-data 86053 Apr  2 09:29 composer.lock
-rw-rw-r-- 1 www-data www-data  2728 Apr  2 09:29 cron.php
drwxrwxr-x 4 www-data www-data  4096 Apr  2 09:29 dev
-rw-rw-r-- 1 www-data www-data  3497 Apr  2 09:29 index.php
drwxrwxr-x 6 www-data www-data  4096 Apr  2 09:29 pub
drwxrwxr-x 2 www-data www-data  4096 Apr  2 09:29 var

```

**Comment by ghost** [comment]
@NiklasBr so this is core issue, run composer install on update folder

**Comment by NiklasBr** [comment]
Didn't help :(
Still no autoload in that folder

**Comment by ghost** [comment]
@NiklasBr you must have update/vendor/autoload.php

**Comment by ghost** [comment]
@NiklasBr so did you have

>  update/vendor/autoload.php 

?

**Comment by NiklasBr** [comment]
No, no autoload.php in that folder before or after running the install command.

**Comment by ghost** [comment]
@NiklasBr that's very strange, in my case composer create project 2.3.1 and i' have .
![DeepinScreenshot_select-area_20190411121856](https://user-images.githubusercontent.com/40756439/55945708-fdb43f00-5c53-11e9-8ea1-fa05cf8383cf.png)

**Comment by NiklasBr** [comment]
After re-installing from scratch it seems that file disappears when we run the command `php bin/magento cron:install`

**Comment by ghost** [comment]
@NiklasBr that very interesting, i'm  will retest

**Comment by ghost** [comment]
@NiklasBr tested several times but not found where command "install" can remove this path.

**Comment by shikhamis11** [comment]
@NiklasBr @engcom-backlog-nazar I am checking this on 2.3.1 version with php 7.2 not getting any issue while cron setup

**Comment by ghost** [comment]
Hi @NiklasBr  i'm closing this issue as this not related to magento codebase, and tor reproducible, if you have some additional steps to reprodcue, feel free to reopen the issue if this one still exists, thanks for collaboration :+1:

**Comment by NiklasBr** [comment]
@engcom-backlog-nazar I can not find *where* either. But now I know for sure that it happens every time we run that command on a new installation.

**Comment by SolsWebdesign** [comment]
Okay, I ran into this as well. Magento 2.3.1 This shop too has no "update/cron.php", no update/vendor directory and no setup upgrade or composer update will me help here.

After a lot of searching around and having a good look at the code base of this shop (that I did not install myself) I have concluded the following:

This shop (which is a production shop!) was not installed the right way (through composer) but directly cloned from the github repository. If the shop is cloned from github (and not installed through the advised composer way) the update/cron.php is missing as well as the update/vendor/* files... (also have a look at issue 1740).

So... since I could not repair this the proper way I threw out the "update/cron.php" from the crontab and now the shop runs fine for now.
